src/main/java/net/rptools/maptool/model/LightSource.java

```diff
@@ -156,37 +173,17 @@ public void setId(@Nonnull GUID id) {
     return name;
   }
 
-  public void setName(@Nonnull String name) {
-    this.name = name;
-  }
-
-  public void add(@Nonnull Light source) {
-    lightList.add(source);
-  }
-
-  public void remove(@Nonnull Light source) {
-    lightList.remove(source);
-  }
-
   /**
-   * @return the lights belonging to this LightSource.
+   * @return A read-only list of lights belonging to this LightSource
    */
   public @Nonnull List<Light> getLightList() {
-    return Collections.unmodifiableList(lightList);
+    return lightList;
```

Comment:
Can we change the type if lightList to ImmutableList? This way it would be more obvious that the list is read only. I can already see somebody adding a constructor that sets the field to 
ArrayList coming.

Reply:
I started with that actually, but had to backtrack. Serialization is our enemy here.

First, modern campaigns will serialize `lightList` in the plainest XML possible (which is great), but older ones, and also our current default light XML, might specify the implementation (such as `LinkedList`). In this case, it's the difference between `<lightList>` and `<lightList class="linked-list">`. If we changed the field type, anything with the latter wouldn't be able to deserialize due to a type mismatch.

Second, XStream only has [special converters](https://x-stream.github.io/converters.html#java.util) for `ArrayList` and `LinkedList`, not for `List` in general. So if we serialize an `ImmutableList`, it will include the implementation details. That will lock us into supporting this 3rd party `List` implementation, and we would have to hope their internals never change in the wrong way.

Both these should are solvable, e.g., by writing custom converters. But it's not trivial and every time I look at them there turns out to be more subtleties than I expect.

Reply:
Can you add a comment to lightList for that?